internal/services/v0/namespace.go

```diff
@@ -184,62 +182,67 @@ func (nss *nsServer) ReadConfig(ctx context.Context, req *v0.ReadConfigRequest)
 }
 
 func (nss *nsServer) DeleteConfigs(ctx context.Context, req *v0.DeleteConfigsRequest) (*v0.DeleteConfigsResponse, error) {
-	headRevision, _ := consistency.MustRevisionFromContext(ctx)
 	ds := datastoremw.MustFromContext(ctx)
 
-	// Ensure that all the specified namespaces can be deleted.
-	for _, nsName := range req.Namespaces {
-		// Ensure the namespace exists.
-		_, _, err := ds.ReadNamespace(ctx, nsName, headRevision)
-		if err != nil {
-			return nil, rewriteNamespaceError(ctx, err)
-		}
+	deletedRevision, err := ds.ReadWriteTx(ctx, func(ctx context.Context, rwt datastore.ReadWriteTransaction) error {
+		// Ensure that all the specified namespaces can be deleted.
+		for _, nsName := range req.Namespaces {
+			// Ensure the namespace exists.
+			_, _, err := rwt.ReadNamespace(ctx, nsName)
+			if err != nil {
+				return err
+			}
 
-		// Check for relationships under the namespace.
-		qy, qyErr := ds.QueryTuples(
-			ctx,
-			&v1.RelationshipFilter{
-				ResourceType: nsName,
-			},
-			headRevision,
-			options.WithLimit(options.LimitOne),
-		)
-		err = shared.ErrorIfTupleIteratorReturnsTuples(
-			ctx,
-			qy,
-			qyErr,
-			"cannot delete definition `%s`, as a relationship exists under it", nsName)
-		if err != nil {
-			return nil, rewriteNamespaceError(ctx, err)
-		}
+			// Check for relationships under the namespace.
+			qy, qyErr := rwt.QueryRelationships(
```

Comment:
Now that I think about it, we might want to reverse these checks: put the `ReadNamespace` below the queries: after all, if there does exist a relationship with that namespace, then we don't need to read it. We only need to read it if there are none

Reply:
Ditto.